yoti-sdk-impl/src/main/java/com/yoti/api/client/spi/remote/call/qrcode/DynamicSharingService.java

```diff
@@ -0,0 +1,89 @@
+package com.yoti.api.client.spi.remote.call.qrcode;
+
+import static com.yoti.api.client.spi.remote.call.HttpMethod.HTTP_POST;
+import static com.yoti.api.client.spi.remote.call.YotiConstants.DEFAULT_CHARSET;
+import static com.yoti.api.client.spi.remote.call.YotiConstants.DEFAULT_YOTI_API_URL;
+import static com.yoti.api.client.spi.remote.call.YotiConstants.PROPERTY_YOTI_API_URL;
+import static com.yoti.api.client.spi.remote.util.Validation.notNull;
+
+import java.io.IOException;
+import java.security.GeneralSecurityException;
+import java.security.KeyPair;
+import java.util.Map;
+
+import com.yoti.api.client.qrcode.DynamicScenario;
+import com.yoti.api.client.qrcode.DynamicShareException;
+import com.yoti.api.client.spi.remote.call.JsonResourceFetcher;
+import com.yoti.api.client.spi.remote.call.ResourceException;
+import com.yoti.api.client.spi.remote.call.ResourceFetcher;
+import com.yoti.api.client.spi.remote.call.UrlConnector;
+import com.yoti.api.client.spi.remote.call.factory.HeadersFactory;
+import com.yoti.api.client.spi.remote.call.factory.PathFactory;
+import com.yoti.api.client.spi.remote.call.factory.SignedMessageFactory;
+
+import com.fasterxml.jackson.databind.ObjectMapper;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public class DynamicSharingService {
+
+    public static DynamicSharingService newInstance() {
```

Comment:
I think that the order of `private/protected/public` its inverted here. Could be 
```
    
    private final PathFactory pathFactory;
    private final HeadersFactory headersFactory;
    private final ObjectMapper objectMapper;
    private final ResourceFetcher resourceFetcher;
    private final SignedMessageFactory signedMessageFactory;

    private final String apiUrl;
    

    public DynamicSharingService(PathFactory pathFactory,
            HeadersFactory headersFactory,
            ObjectMapper objectMapper,
            ResourceFetcher resourceFetcher,
            SignedMessageFactory signedMessageFactory) {
        this.pathFactory = pathFactory;
        this.headersFactory = headersFactory;
        this.objectMapper = objectMapper;
        this.resourceFetcher = resourceFetcher;
        this.signedMessageFactory = signedMessageFactory;

        apiUrl = System.getProperty(PROPERTY_YOTI_API_URL, DEFAULT_YOTI_API_URL);
    }
    
    public static DynamicSharingService newInstance() {
        return new DynamicSharingService(
                new PathFactory(),
                new HeadersFactory(),
                new ObjectMapper(),
                JsonResourceFetcher.newInstance(),
                SignedMessageFactory.newInstance());
    }
```
But it's up to you. Also the consturctore should be private if we are using the `newInstance()`

Reply:
The constructor should be private, fixed.
I'm going to leave the static code at the top.